packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * External dependencies
+ */
+import { createElement, Fragment } from 'react';
+
+let indoc,
+	offset,
+	output,
+	stack,
+	keyIndex;
+
+/**
+ * Matches tags in the localized string
+ *
+ * This is used for extracting the tag pattern groups for parsing the localized
+ * string and along with the map converting it to a react element.
+ *
+ * There are four references extracted using this tokenizer:
+ *
+ * match: Full match of the tag (i.e. <strong>, </strong>, <br/>)
+ * isClosing: If the match has a closing tag (i.e. </)
+ * name: The name portion of the tag (strong, br) (if )
+ * isSelfClosed: Whether the match is for a self-closing tag (i.e. />)
+ *
+ * @type RegExp
+ */
+const tokenizer = /<(\/)?(\w+)\s*(\/)?>/g;
+
+/**
+ * This receives the value for a map element which consists of an element and
+ * it's props and returns a element creator function.
+ *
+ * @param {Array} [ element, props ] The first item in the array is expected to
+ *                                   be a string or a react component. The
+ *                                   second item is expected to be either
+ *                                   undefined or an object.
+ */
+const elementCreator = ( [ element, props = {} ] ) => ( children ) => {
+	if ( typeof element !== 'string' && typeof element !== 'function' ) {
+		throw new Error( `Not a valid element (${ element })` );
+	}
+	props.key = ++keyIndex;
+	return createElement( element, props, children );
+};
+
+/**
+ * An object describing a component to be created.
+ *
+ * This is used by the string iterator to track children that get added to an
+ * element when it is created. This allows for collecting nested elements in
+ * the string before creating the parent.
+ *
+ * @param   {Function}  creator An element creator that will be invoked for
+ *                              actually creating the react element with the
+ *                              provided children.
+ * @param   {Array}  children   Array of children to be provided to the element
+ *                              creator when invoked.
+ *
+ * @return  {Object}            An object returning the creator and children.
+ */
```

Comment:
It'd be worth adding `@{internal}` here (or whatever that syntax is supposed to be.

this function isn't truly necessary but it's here because I added it to the default Gutenberg parser as an optimization. by creating our object as a constructor function the JS engine will create a hidden class and bypass the expensive dynamic elements of object creation.

in the translation function it's probably still worth being here but I'd bet isn't nearly as important. still, it's helpful in the code to serve as pseudo-static typing. we could measure the performance and know for sure.

all-in-all, like `Frame` below, this is an internal element of the algorithm placed as an optimization

Reply:
addressed in 79081c8